lib/node_modules/@stdlib/lapack/base/dlaqge/lib/base.js

```diff
@@ -0,0 +1,130 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable max-len, max-params */
+
+'use strict';
+
+// MODULES //
+
+var isRowMajor = require( '@stdlib/ndarray/base/assert/is-row-major' );
+
+
+// MAIN //
+
+/**
+* Equilibrates a general `M` by `N` matrix `A` using the row and scaling factors in vectors `R` and `C`.
+*
+* @private
+* @param {NonNegativeInteger} M - number of rows in matrix `A`
+* @param {NonNegativeInteger} N - number of columns in matrix `A`
+* @param {Float64Array} A - input matrix
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index for `A`
+* @param {Float64Array} R - row scale factors
+* @param {NonNegativeInteger} strideR - stride length for `R`
+* @param {NonNegativeInteger} offsetR - starting index of `R`
+* @param {Float64Array} C - column scale factors
+* @param {NonNegativeInteger} strideC - stride length for `C`
+* @param {NonNegativeInteger} offsetC - starting index of `C`
+* @param {number} rowcnd - ratio of the smallest to the largest row scale factor
+* @param {number} colcnd - ratio of the smallest to the largest column scale factor
+* @param {number} amax - absolute value of the largest matrix entry
+* @returns {string} equilibration factor
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+* var R = new Float64Array( [ 1.0, 11.0 ] );
+* var C = new Float64Array( [ 1.0, 11.0 ] );
+*
+* var out = dlaqge( 2, 2, A, 2, 1, 0, R, 1, 0, C, 1, 0, 1.0/11.0, 1.0/11.0, 4.0 );
+* // out => 'B'
+* // A => <Float64Array>[ 1.0, 22.0, 33.0, 484.0 ]
+*/
+function dlaqge( M, N, A, strideA1, strideA2, offsetA, R, strideR, offsetR, C, strideC, offsetC, rowcnd, colcnd, amax ) {
+	var thresh;
+	var large;
+	var small;
+	var sa0;
+	var sa1;
+	var rj;
+	var ia;
+	var i;
+	var j;
+
+	if ( M <= 0 || N <= 0 ) {
+		return 'N';
+	}
+
+	if ( isRowMajor( [ strideA1, strideA2 ] ) ) {
```

Comment:
Given that `A` is `MxN`, you also need to swap the loop limits when performing loop interchange. It isn't enough to simply swap the strides.

Reply:
I do not remember what I did. IIRC the code is already optimized, `sa0` was used internally for both `row-major` and `column-major` and thus I removed code duplication.